File: practica stuff/C/HelloTimeOfDay.c
#include <stdio.h>
#include <sys/time.h>
#include <time.h>
#include <stdlib.h>

int main() {
  char name[32];
  struct timeval tv;

  printf("Your name please: ");
  scanf("%31s", name);
 
  printf("\n");
 
  if(gettimeofday(&tv, NULL) == 0) {
    struct tm *time = localtime( &(tv.tv_sec) );
    printf("%d\n", time->tm_hour ); //just extra stuff
    if(time->tm_hour <= 12 ) {
      printf("Good Morning %s!\n", name);
      system("pause");
      return 0;
 
    } else if(time->tm_hour> 12 && time->tm_hour<= 18 ) {
      printf("Good Afternoon %s!\n", name);
      system("pause");
      return 0;
 
    } else {
      printf("Good Evening %s!\n", name);
      system("pause");
      return 0;
    }
 
  } else {
    printf("Could not get time of day");
    system("pause");
    return 1;
  }
  system("pause");
  return 2;
}

File: practica stuff/Java/HelloTimeOfDay.java
import java.util.Scanner;
import java.time.LocalTime;
import java.time.*; //get everything(?) from [time] (not sure if works, needs testing)
import java.time.format.DateTimeFormatter; // more info at https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html

public class HelloTimeOfDay { // public class [filename]
    public static void main(String[] args)
    {
        DateTimeFormatter dtf = DateTimeFormatter.ofPattern("H");
        DateTimeFormatter twelvehour = DateTimeFormatter.ofPattern("hh");
        int timeLabel12 = Integer.parseInt(LocalTime.now().format(twelvehour));
        int timeLabel = Integer.parseInt(LocalTime.now().format(dtf));
        Scanner nameinput = new Scanner(System.in);
        System.out.print("Input your first name: ");
        String firstname = nameinput.next();
        System.out.print("Input your last name: ");
        String lastname = nameinput.next();
        System.out.println();
        System.out.print("Hello " + firstname + " " + lastname + "! It's " + timeLabel12 + " in the "); //timeLabel12 for AM&PM, timeLabel for 24h.
        if (timeLabel >= 7 && timeLabel < 13) { //Use timeLabel to recognize Morning/Afternoon/Evening easier than 12h format.
            System.out.print("Morning!");
        } else if (timeLabel >= 13 && timeLabel < 18) {
            System.out.print("Afternoon!");
        } else {
            System.out.print("Evening!");
        }
        // Line 19: double ++ ONLY between " ".
    }
}